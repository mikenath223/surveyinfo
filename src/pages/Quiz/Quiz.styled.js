import styled from 'styled-components';
import { breakpoints } from 'styles/global';

export const Wrapper = styled.main`
  min-height: 100vh;
  background-color: ${({ theme }) => theme.colors.bottomDropBg};
`

export const Hero = styled.section`
  background: url(/assets/survey.png) center no-repeat;
  background-color: ${({ theme }) => theme.colors.headBg};
  background-position: 90%;
  background-size: 300px;
  padding: 30px 0 0 100px;
  min-height: 300px;

  @media ${breakpoints.sm} {
    background-size: 150px;
    padding: 30px 0 0 30px;
  }
`;

export const HeroText = styled.h1`
  font-size: ${({ theme }) => theme.textSizes.mainText};
  color: ${({ theme }) => theme.colors.mainText};
  font-family: Montserrat;
  font-weight: 700;
  padding-top: 4.5rem;
  text-transform: uppercase;
  margin: 0;

  @media ${breakpoints.md} {
    font-size: 20px;
    padding-top: 7.5rem;
  }
`;

export const HeroSubText = styled.h4`
  font-size: ${({ theme }) => theme.textSizes.subText};
  color: ${({ theme }) => theme.colors.subText};
  font-style: italic;

  @media ${breakpoints.sm} {
    font-size: 14px;
  }
`;

export const DashBoard = styled.section`
  margin: -90px auto 100px auto;
  min-height: 500px;
  max-width: 1200px;
  padding: 20px;
`;

export const InnerSection = styled.div`
  background-color: ${({ theme }) => theme.colors.dashboardBg};
  display: flex;
  max-width: 1200px;
  min-height: 500px;
  border-radius: 20px;
`;

export const BoardAside = styled.div`
  background: url(/assets/mailbox.png) center no-repeat;
  background-color: ${({ theme }) => theme.colors.asideBg};
  background-size: 190px;
  width: 500px;
  min-height: 500px;
  border-radius: 20px 0;

  @media ${breakpoints.md} {
    display: none;
  }
`;

export const FormSection = styled.div`
  display: flex;
  justify-content: center;
  width: 100%
`;