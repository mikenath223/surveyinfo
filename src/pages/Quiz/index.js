import React from 'react';
import { connect } from 'react-redux';
import { useHistory } from 'react-router-dom';
import { withTheme } from 'styled-components';
import {
  Hero,
  HeroText,
  HeroSubText,
  DashBoard,
  InnerSection,
  BoardAside,
  FormSection,
} from './Quiz.styled';
import { setThemeMode, saveQuizResult } from 'store/actions';
import Form from 'components/Form';

const Quiz = ({ theme, setThemeMode, saveQuizResult }) => {
  const history = useHistory();

  const handleSubmit = values => {
    const { message, date, layout, color, text } = values;
    saveQuizResult({ message, date });
    const theme = `${color}_${layout}_${text}_TEXT`;
    setThemeMode(theme);
    history.push('/result');
  }

  return <main>
    <Hero>
      <HeroText>Survey Grill</HeroText>
      <HeroSubText>We invite you to come take our survey</HeroSubText>
    </Hero>
    <DashBoard>
      <InnerSection>
        <BoardAside />
        <FormSection>
          <Form handleSubmit={handleSubmit} />
        </FormSection>
      </InnerSection>
    </DashBoard>
  </main>
}

const mapDispatchToProps = {
  saveQuizResult,
  setThemeMode
}

export default withTheme(connect(null, mapDispatchToProps)(Quiz))