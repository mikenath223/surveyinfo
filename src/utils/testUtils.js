import React from 'react';
import { ThemeProvider } from 'styled-components';
import { render } from '@testing-library/react';
import { theme } from 'styles/global';

export const withThemeProvider = (component) =>
  render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);

export const withResultThemeProvider = (component) =>
  render(<ThemeProvider theme={theme}>
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
  </ThemeProvider>)