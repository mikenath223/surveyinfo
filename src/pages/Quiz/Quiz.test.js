import React from 'react';
import 'jest-styled-components';
import { withReduxThemeProvider } from 'utils/testUtils';
import Quiz from './index';

describe('Quiz page', function () {
  it('should match snapshot', () => {
    const { container } = withReduxThemeProvider(<Quiz />);

    expect(container).toMatchSnapshot();
  });
});
