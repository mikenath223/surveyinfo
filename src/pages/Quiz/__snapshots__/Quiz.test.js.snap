// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quiz page should match snapshot 1`] = `
.c0 {
  min-height: 100vh;
  background-color: #EBEBED;
}

.c1 {
  background: url(/assets/survey.png) center no-repeat;
  background-color: #2B13AF;
  background-position: 90%;
  background-size: 300px;
  padding: 30px 0 0 100px;
  min-height: 300px;
}

.c2 {
  color: #fff;
  font-family: Montserrat;
  font-weight: 700;
  padding-top: 4.5rem;
  text-transform: uppercase;
  margin: 0;
}

.c3 {
  color: #999CE0;
  font-style: italic;
}

.c4 {
  margin: -90px auto 100px auto;
  min-height: 500px;
  max-width: 1200px;
  padding: 20px;
}

.c5 {
  background-color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  max-width: 1200px;
  min-height: 500px;
  border-radius: 20px;
}

.c6 {
  background: url(/assets/mailbox.png) center no-repeat;
  background-color: silver;
  background-size: 190px;
  width: 500px;
  min-height: 500px;
  border-radius: 20px 0;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
}

.c8 {
  font-weight: 600;
  text-align: center;
  line-height: 2.5;
  margin-bottom: 50px;
}

.c9 {
  width: 80%;
  margin: auto;
}

.c11 {
  text-align: center;
  margin: 1.5rem 0;
  font-size: 15px;
}

.c10 {
  border-top: 1px solid grey;
  margin-top: 4rem;
  margin-bottom: 0;
  opacity: 0.3;
}

@media (max-width:576px) {
  .c1 {
    background-size: 150px;
    padding: 30px 0 0 30px;
  }
}

@media (max-width:768px) {
  .c2 {
    font-size: 20px;
    padding-top: 7.5rem;
  }
}

@media (max-width:576px) {
  .c3 {
    font-size: 14px;
  }
}

@media (max-width:768px) {
  .c6 {
    display: none;
  }
}

<div>
  <main
    class="c0"
  >
    <section
      class="c1"
    >
      <h1
        class="c2"
      >
        Survey Grill
      </h1>
      <h4
        class="c3"
      >
        We invite you to come take our survey
      </h4>
    </section>
    <section
      class="c4"
    >
      <div
        class="c5"
      >
        <div
          class="c6"
        />
        <div
          class="c7"
        >
          <form
            class="ml-lg-5 m-4"
            novalidate=""
          >
            <h2
              class="c8"
            >
              Take the survey
            </h2>
            <div
              class="pb-4 mx-0 form-group row"
            >
              <label
                class="form-label"
                for="validationFormikMessage1"
              >
                Message
              </label>
              <div
                class="input-group"
              >
                <textarea
                  class="mh-100 form-control"
                  data-testid="messageInput"
                  id="validationFormikMessage1"
                  name="message"
                  rows="7"
                  spellcheck="true"
                />
                <div
                  class="invalid-tooltip"
                  data-testid="messageError"
                />
                <div
                  class="valid-tooltip"
                >
                  Looks good!
                </div>
              </div>
            </div>
            <div
              class="pb-4 mx-0 form-group row"
            >
              <label
                class="col-12 pl-0 form-label"
                for="validationFormikDate2"
              >
                Date
              </label>
              <div
                class="p-0 col-5 input-group"
              >
                <input
                  class="form-control"
                  id="validationFormikDate2"
                  name="date"
                  type="date"
                  value=""
                />
                <div
                  class="valid-tooltip"
                >
                  Looks good!
                </div>
                <div
                  class="invalid-tooltip"
                />
              </div>
            </div>
            <div
              class="pb-4 px-0 mx-0 mt-4 form-row"
            >
              <div
                class="col-12 pl-0 form-label col"
              >
                Customize your theme
              </div>
              <div
                class="col-12 mx-0 border border-1 mt-1 p-2 row"
              >
                <div
                  class="form-group col-md-4"
                >
                  <label
                    class="font-weight-bold form-label"
                  >
                    Layout
                  </label>
                  <div
                    class="input-group"
                  >
                    <div
                      aria-labelledby="my-radio-group"
                      role="group"
                    >
                      <div
                        class="form-check"
                      >
                        <input
                          checked=""
                          class="form-check-input"
                          id="layout"
                          name="layout"
                          type="radio"
                          value="LIST"
                        />
                        <label
                          class="form-check-label"
                          for="layout"
                          title=""
                        >
                          List
                        </label>
                      </div>
                      <div
                        class="form-check"
                      >
                        <input
                          class="form-check-input"
                          id="layout"
                          name="layout"
                          type="radio"
                          value="GRID"
                        />
                        <label
                          class="form-check-label"
                          for="layout"
                          title=""
                        >
                          Grid
                        </label>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="form-group col-md-4"
                >
                  <label
                    class="font-weight-bold form-label"
                  >
                    Color Mode
                  </label>
                  <div
                    class="input-group"
                  >
                    <div
                      aria-labelledby="my-radio-group"
                      role="group"
                    >
                      <div
                        class="form-check"
                      >
                        <input
                          checked=""
                          class="form-check-input"
                          id="color"
                          name="color"
                          type="radio"
                          value="LIGHT"
                        />
                        <label
                          class="form-check-label"
                          for="color"
                          title=""
                        >
                          Light Mode
                        </label>
                      </div>
                      <div
                        class="form-check"
                      >
                        <input
                          class="form-check-input"
                          id="color"
                          name="color"
                          type="radio"
                          value="DARK"
                        />
                        <label
                          class="form-check-label"
                          for="color"
                          title=""
                        >
                          Dark Mode
                        </label>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="form-group col-md-4"
                >
                  <label
                    class="font-weight-bold form-label"
                  >
                    Text Size
                  </label>
                  <div
                    class="input-group"
                  >
                    <div
                      aria-labelledby="my-radio-group"
                      role="group"
                    >
                      <div
                        class="form-check"
                      >
                        <input
                          checked=""
                          class="form-check-input"
                          id="text"
                          name="text"
                          type="radio"
                          value="REGULAR"
                        />
                        <label
                          class="form-check-label"
                          for="text"
                          title=""
                        >
                          Regular
                        </label>
                      </div>
                      <div
                        class="form-check"
                      >
                        <input
                          class="form-check-input"
                          id="text"
                          name="text"
                          type="radio"
                          value="LARGE"
                        />
                        <label
                          class="form-check-label"
                          for="text"
                          title=""
                        >
                          Large
                        </label>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="col-12 mx-0 my-5 row"
            >
              <button
                class="mx-auto btn btn-primary"
                type="submit"
              >
                Submit survey
              </button>
            </div>
          </form>
        </div>
      </div>
    </section>
    <div
      class="c9 row"
    >
      <div
        class="col-sm-12"
      >
        <hr
          class="c10"
        />
      </div>
      <div
        class="col-sm-12"
      >
        <div
          class="c11"
        >
          © 2020 CheabanTest. All rights reserved.
        </div>
      </div>
    </div>
  </main>
</div>
`;
